steps/25_dataset_collection/step.md

```diff
@@ -0,0 +1,158 @@
+from sktime.classification.dummy import DummyClassifier
+from sktime.dataset_collection import DatasetCollection # To implement
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import KFold
+
+
+benchmark = ClassificationBenchmark()
+
+benchmark.add_estimator(
+    estimator=DummyClassifier(),
+    estimator_id="DummyClassifier",
+)
+
+cv_splitter = KFold(n_splits=3)
+scorers = [accuracy_score]
+
+# Instantiate a dataset collection, each collection will have a different class following strategy pattern, e.g., collection of datasets that were used in TSC Bakeoff paper, returns a list of datasets loaded as sktime dataset containers
+datasets = DatasetCollection("TSCBakeOff")
+
+# If we want an additional dataset to extend the study, let's assume ArrowHead was not present in the initial study but we want to add it to our collection for this particular study
+from sktime.datasets import load_arrow_head
+datasets.add(load_arrow_head)
+
+for dataset in datasets:
+    benchmark.add_task(
+        dataset,
+        cv_splitter,
+        scorers,
+    )
+
+results_df = benchmark.run("./classification_results.csv")
+results_df.T
+```
+### Proposed Classes
+
+#### `BaseDatasetCollection` Class
+
+```python
+class BaseDatasetCollection(BaseObject):
+        """Base class for dataset collection strategies using sktime's BaseObject."""
+    
+    def __init__(self):
+        super().__init__()
+    
+    def get_datasets(self):
+        """Return a list of dataset loading functions."""
+        raise NotImplementedError("Subclasses must implement get_datasets()")
+    
+    def get_collection_name(self):
+        """Return the name of the collection."""
+        raise NotImplementedError("Subclasses must implement get_collection_name()")
+```
+
+#### `TSCBakeOffCollection` Example
+```python
+class TSCBakeOffCollection(BaseDatasetCollection):
+    """Collection of TSC Bake-off datasets."""
+    
+    def __init__(self, subset=None):
+        """
+        Initialize TSC Bake-off collection.
+        """
+        self.subset = subset
+        super().__init__()
+    
+    def get_collection_name(self):
+        return "TSC Bake-off Collection"
+    
+    def get_datasets(self):
+        """Return a list of dataset loaders from the TSC Bake-off paper."""
+        from sktime.datasets import load_UEA_UCR_dataset
+
+        dataset_names = [] # Contributor writing the collection will have to manually add the datasets used in the experiment
+        all_datasets = []
+        for name in dataset_names:
+            # If UEA UCR archive has all the datasets then load from there or use additional loaders, even custom ones
+            dataset = load_UEA_UCR_dataset(name=name)
+            all_datasets.append(dataset)
+        
+        return all_datasets
+```
+
+```python
+
+class DatasetCollection(BaseObject):
+    """
+    A collection of datasets following the strategy pattern.
+    """
+
+    _collections = {
```

Comment:
this design is not very extensible!

Why: for adding a collection, you need to:

1. add a new object
2. add it to this register

Generally, designs requiring a registry are less extensible than designs using the strategy pattern only.